interface/frontend/src/components/open-image/AreaSelect.vue

```diff
@@ -0,0 +1,271 @@
+        } else if (this.draggingEntireArea && lastLocation) {
+          // if entire area is being dragged - update all points
+
+          for (let pointLocation of this.areaCoordinates) {
+            pointLocation[0] += dX
+            pointLocation[1] += dY
+          }
+        } else {
+          return
+        }
+
+        this.anyMovesDetected = true
+        this.redraw()
+      },
+      onMouseUp (e) {
+        if (this.draggingPointIndex < 0 && !this.draggingEntireArea) {
+          return
+        }
+
+        if (this.anyMovesDetected) {
+          this.notify()
+        }
+
+        this.draggingPointIndex = -1
+        this.draggingEntireArea = false
+        this.anyMovesDetected = false
+      },
+      redrawAndNotify () {
+        this.redraw()
+        this.notify()
+      },
+      notify () {
+        this.$emit('selection-changed', this.areaCoordinates)
+      },
+      redraw () {
+        this.canvasContext.clearRect(0, 0, this.canvas.width, this.canvas.height)
+        this.canvasContext.globalCompositeOperation = 'source-over'
+        this.drawArea()
+      },
+      drawArea () {
+        let coords = this.areaCoordinates
+
+        if (coords.length < 1) {
+          return false
+        }
+
+        this.canvasContext.lineWidth = 1
+
+        // draw polygon
+        this.canvasContext.beginPath()
+        this.canvasContext.moveTo(coords[0][0], coords[0][1])
+
+        for (let pointLocation of coords) {
+          this.canvasContext.lineTo(pointLocation[0], pointLocation[1])
+        }
+
+        // finish the polygon
+        this.canvasContext.closePath()
+        this.canvasContext.strokeStyle = strokePolygonRgb
+        this.canvasContext.stroke()
+        this.canvasContext.fillStyle = fillPolygonRgba
+        this.canvasContext.fill()
+
+        // draw draggable points
+        this.canvasContext.strokeStyle = strokeCoordRgb
+        this.canvasContext.fillStyle = fillCoordRgb
+
+        for (let pointLocation of coords) {
+          this.canvasContext.beginPath()
+          this.canvasContext.arc(pointLocation[0], pointLocation[1], 4, 0, 2 * Math.PI)
+          this.canvasContext.fill()
+        }
+      },
+      getElementOffset (element) {
+        const de = document.documentElement
+        const box = element.getBoundingClientRect()
+        const top = box.top + window.pageYOffset - de.clientTop
+        const left = box.left + window.pageXOffset - de.clientLeft
+        return {top: top, left: left}
+      },
+      preventDefault (e) {
+        if (e.stopPropagation) e.stopPropagation()
+        if (e.preventDefault) e.preventDefault()
+      },
+      getMouseLocation (e) {
+        const canvasOffset = this.getElementOffset(this.canvas)
+        const x = (e.pageX - canvasOffset.left)
+        const y = (e.pageY - canvasOffset.top)
+
+        return {x, y}
+      },
+      pointToLineLength (x, y, x0, y0, x1, y1, o) {
```

Comment:
Dat math. Was this code, err, borrowed from somewhere? If so, we should really credit it :)

Reply:
For `pointToLineLength` it was used by [original repo](https://github.com/neshte/jquery-canvas-area-draw) without mentioning of the author. But I have found, that this method comes [from here](http://jsfromhell.com/math/dot-line-length). Will include the reference.
My other question remains open - should we add the license and reference to [original repo](https://github.com/neshte/jquery-canvas-area-draw)?